test/integration/client/client_test.go

```diff
@@ -1342,3 +1350,112 @@ func TestExtractModifyApply_ForceOwnership(t *testing.T) {
 		t.Errorf("createMgrExtracted apply configuration did not match expected, got:\n%s\n", cmp.Diff(expectedCreateExtracted, createMgrExtracted))
 	}
 }
+
+// regression test for https://issues.k8s.io/117258
+func TestAPIServerTransportMetrics(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, "AllAlpha", true)()
+	defer featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, "AllBeta", true)()
+
+	result := kubeapiservertesting.StartTestServerOrDie(t, nil, []string{"--disable-admission-plugins", "ServiceAccount"}, framework.SharedEtcd())
+	defer result.TearDownFn()
+
+	client := clientset.NewForConfigOrDie(result.ClientConfig)
+
+	hits1, misses1, entries1 := checkTransportMetrics(t, client)
```

Comment:
does this see hits/misses/entry counts from all tests that ran in this package, or just this test instance?

Reply:
this is an interesting question, it performs a curl to the apiserver to not use the global registry

> 	body, err := client.RESTClient().Get().AbsPath("/metrics").DoRaw(context.Background())

but since all will be compiled in the same binary I assume that all apiservers running will share the same registry for metrics ... more reasons to operate on statistics and avoid running test in parallel

Reply:
can we at least reset the metrics at the start of the test, to avoid other tests in the package impacting what we see?

Reply:
done